### PR TITLE
Add tests for boundary conditions when computing ISO week number

### PR DESCRIPTION
**Pull request by lopopolo**
These tests were pulled from the following PR in chrono:

- https://github.com/chronotope/chrono/pull/966

Add a single test with these boundary condition dates. The dates were validated with MRI Ruby v3.1.2. The third test case differs between chrono and MRI, but since we target MRI, use it as the source of truth.

No logic was changed as part of this commit, only additional tests were added.

### CODE REVIEW

**Comment by lopopolo** [comment]
Hi @x-hgg-x, I tried to dig into the `week_number` and `iso_week_number` functions to see what's going wrong but I don't quite understand all of the tricky math. Would you be able to take a look?

File: src/tests/format.rs [hunk 1]
```diff
@@ -922,3 +922,34 @@ fn test_format_copies_non_specifier_bytes() {
         ],
     );
 }
+
+#[test]
+fn test_chrono_pr_966_week_numbers() {
+    let times = [
+        MockTime::new(2020, 1, 12, 0, 0, 0, 0, 4, 1, 0, false, 0, ""),
+        MockTime::new(2019, 1, 13, 0, 0, 0, 0, 4, 1, 0, false, 0, ""),
+        MockTime::new(2007, 12, 31, 0, 0, 0, 0, 4, 1, 0, false, 0, ""),
```
Comment (x-hgg-x) [inline src/tests/format.rs]:
```suggestion
        MockTime::new(2020, 1, 12, 0, 0, 0, 0, 0, 12, 0, false, 0, ""),
        MockTime::new(2019, 1, 13, 0, 0, 0, 0, 0, 13, 0, false, 0, ""),
        MockTime::new(2007, 12, 31, 0, 0, 0, 0, 1, 365, 0, false, 0, ""),
```

File: src/tests/format.rs [hunk 2]
```diff
@@ -922,3 +922,34 @@ fn test_format_copies_non_specifier_bytes() {
...
+            // ```
+            // [3.1.2] > Time.utc(2019, 1, 13).strftime("%Y-%W-%w")
+            // => "2019-01-0"
+            // ```
+            "2019-01-0",
+            // ```
+            // [3.1.2] > Time.utc(2007, 12, 31).strftime("%Y-%W-%w")
+            // => "2007-53-1"
+            // ```
+            "2007-12-52",
```
Comment (x-hgg-x) [inline src/tests/format.rs]:
```suggestion
            "2007-53-1",
```

**Review by x-hgg-x — CHANGES_REQUESTED** [review]

**Comment by x-hgg-x** [comment]
Since you are testing multiple formatting specifiers at the same time, you must ensure that all fields of the `MockTime` are consistent and represent a valid date and time. We are currently doing light checks in the `CheckedTime` trait, but it is only to prevent panics due to overflow, and we don't validate the consistency of all fields.

**Comment by lopopolo** [comment]
It looks like that did the trick! Thanks for the help 🙇

**Review by x-hgg-x — APPROVED** [review]

**Comment by x-hgg-x** [comment]
Please remember to squash the commits and modify their description, since we didn't actually change any code.
